visualization/app/codeCharta/services/loadInitialFile/loadInitialFile.service.ts

```diff
@@ -25,50 +71,351 @@ export class LoadInitialFileService {
+				break
+			case "markedPackages":
+				this.store.dispatch(setMarkedPackages({ value }))
+				break
+			default: {
+				const exhaustiveCheck: never = key
+				throw new Error(`Unhandled key: ${exhaustiveCheck}`)
+			}
+		}
+	}
+
+	private mapDynamicSettingToAction(key: keyof DynamicSettings, value: any) {
+		switch (key) {
+			case "areaMetric":
+				this.store.dispatch(setAreaMetric({ value }))
+				break
+			case "heightMetric":
+				this.store.dispatch(setHeightMetric({ value }))
+				break
+			case "edgeMetric":
+				this.store.dispatch(setEdgeMetric({ value }))
+				break
+			case "colorMetric":
+				this.store.dispatch(setColorMetric({ value }))
+				break
+			case "colorMode":
+				this.store.dispatch(setColorMode({ value }))
+				break
+			case "sortingOption":
+				this.store.dispatch(setSortingOption({ value }))
+				break
+			case "colorRange":
+				this.store.dispatch(setColorRange({ value }))
+				break
+			case "distributionMetric":
+				this.store.dispatch(setDistributionMetric({ value }))
+				break
+			case "focusedNodePath":
+				this.store.dispatch(setAllFocusedNodes({ value }))
+				break
+			case "searchPattern":
+				this.store.dispatch(setSearchPattern({ value }))
+				break
+			case "margin":
+				this.store.dispatch(setMargin({ value }))
+				break
+			default: {
+				const exhaustiveCheck: never = key
+				throw new Error(`Unhandled key: ${exhaustiveCheck}`)
+			}
+		}
+	}
+
+	private mapAppSettingToAction(key: keyof AppSettings, value: any) {
+		switch (key) {
+			case "amountOfTopLabels":
+				this.store.dispatch(setAmountOfTopLabels({ value }))
+				break
+			case "amountOfEdgePreviews":
+				this.store.dispatch(setAmountOfTopLabels({ value }))
+				break
+			case "edgeHeight":
+				this.store.dispatch(setEdgeHeight({ value }))
+				break
+			case "scaling":
+				this.store.dispatch(setScaling({ value }))
+				break
+			case "hideFlatBuildings":
+				this.store.dispatch(setHideFlatBuildings({ value }))
+				break
+			case "invertHeight":
+				this.store.dispatch(setInvertHeight({ value }))
+				break
+			case "invertArea":
+				this.store.dispatch(setInvertArea({ value }))
+				break
+			case "isWhiteBackground":
+				this.store.dispatch(setIsWhiteBackground({ value }))
+				break
+			case "mapColors":
+				this.store.dispatch(setMapColors({ value }))
+				break
+			case "isPresentationMode":
+				this.store.dispatch(setPresentationMode({ value }))
+				break
+			case "showOnlyBuildingsWithEdges":
+				this.store.dispatch(setShowOnlyBuildingsWithEdges({ value }))
+				break
+			case "isEdgeMetricVisible":
+				this.store.dispatch(setIsEdgeMetricVisible({ value }))
+				break
+			case "resetCameraIfNewFileIsLoaded":
+				this.store.dispatch(setResetCameraIfNewFileIsLoaded({ value }))
+				break
+			case "isLoadingMap":
+				this.store.dispatch(setIsLoadingMap({ value }))
+				break
+			case "isLoadingFile":
+				this.store.dispatch(setIsLoadingFile({ value }))
+				break
+			case "sortingOrderAscending":
```

Comment:
I added a comment explaining that those properties not applied on purpose. If we were to apply the sorting-order and the pin-state we would also have to save which folders were opened / collapsed in the file-explorer. Because this would have meant a lot of effort, we decided to ignore all propeties belonging to the file-explorer.